src/navigator/NavigationService.ts

```diff
@@ -196,6 +196,12 @@ export async function isScreenOnForeground(screen: Screens) {
   return activeRouteState?.routes[activeRouteState?.routes.length - 1]?.name === screen
 }
 
+export async function isBottomSheetVisible(screen: Screens) {
+  await ensureNavigator()
+  const state = navigationRef.current?.getRootState()
+  return !!state?.routes.find((route) => route.name === screen)
+}
+
```

Comment:
Could we use `isScreenOnForeground` instead?

Reply:
we discussed this already, noting it here for reference - `isScreenOnForeground` doesn't work here, it does some pretty involved things to check the routes which is needed for the stack navigator but doesn't apply for the root navigator anymore (it always returns false whether a bottom sheet screen is there or not)

the `state.routes` will either contain the bottom sheet screen or not, there's no need to dig deeper into the state. hopefully the lib will introduce a better way to do this soon and we can swap this function out :)